ogzaf/demographics.py

```diff
@@ -1,96 +1,352 @@
-    # within the JSON structure
-    df = pd.json_normalize(j["data"])
-    # Loop until there are new pages with data
-    while j["nextPage"] is not None:
-        # Reset the target to the next page
-        target = j["nextPage"]
-        # call the API for the next page
-        response = requests.get(target)
-        # Convert response to JSON format
-        j = response.json()
-        # Store the next page in a data frame
-        df_temp = pd.json_normalize(j["data"])
-        # Append next page to the data frame
-        df = df.append(df_temp)
-
-    return df
+    if end_year is None:
+        end_year = start_year
+    # UN variable code for Population by 1-year age groups and sex
+    pop_code = "47"
+    # UN variable code for Fertility rates by age of mother (1-year)
+    fert_code = "68"
+
+    if download:
+        pop_target = (
+            "https://population.un.org/dataportalapi/api/v1/data/indicators/"
+            + pop_code
+            + "/locations/"
+            + country_id
+            + "/start/"
+            + str(start_year)
+            + "/end/"
+            + str(end_year)
+            + "?format=csv"
+        )
+        fert_target = (
+            "https://population.un.org/dataportalapi/api/v1/data/indicators/"
+            + fert_code
+            + "/locations/"
+            + country_id
+            + "/start/"
+            + str(start_year)
+            + "/end/"
+            + str(end_year)
+            + "?format=csv"
+        )
+    else:
+        pop_target = os.path.join(DATA_DIR, "un_zaf_pop.csv")
+        fert_target = os.path.join(DATA_DIR, "un_zaf_fert.csv")
+
+    # Convert .csv file to Pandas DataFrame
+    pop_df = pd.read_csv(
+        pop_target,
+        sep="|",
+        header=1,
+        usecols=["TimeLabel", "SexId", "Sex", "AgeMid", "Value"],
+        float_precision="round_trip",
+    )
+    fert_rates_df = pd.read_csv(
+        fert_target,
+        sep="|",
+        header=1,
+        usecols=["TimeLabel", "AgeMid", "Value"],
+        float_precision="round_trip",
+    )
+
+    # Rename variables in the population and fertility rates data
+    pop_df.rename(
+        columns={
+            "TimeLabel": "year",
+            "SexId": "sex_num",
+            "Sex": "sex_str",
+            "AgeMid": "age",
+            "Value": "pop",
+        },
+        inplace=True,
+    )
+    fert_rates_df.rename(
+        columns={
+            "TimeLabel": "year",
+            "AgeMid": "age",
+            "Value": "births_p_1000f",
+        },
+        inplace=True,
+    )
+
+    # Clean the data
+    # I don't know why in the pop_df population data by age and sex and year
+    # there are 10 different population numbers for each sex and age and year
+    # and all the other variables are equal. I just average them here.
+    pop_df = (
+        pop_df.groupby(["year", "sex_num", "sex_str", "age"])
+        .mean()
+        .reset_index()
+    )
```

Comment:
I don't have this problem. I only see a duplicate for ages 0 and 1 and when using the `AgeMid` variable (see proposed changes above to correct this).

Reply:
@SeaCelo. Your `AgeStart` change fixed this problem. Nice catch. I have updated the code such that these lines are deleted because they are no longer necessary.